### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by bruk19**
In this PR:-

- Created User model classes
- Created Comment model classes
- Created Post model classes
- Created Like model classes
- Set up associations between models
- Added In User the following method:
     A method that returns the 3 most recent posts for a given user.
- Added In Post the following method:
    A method that returns the 5 most recent comments for a given post.
- Added In Comment the following method:
    A method that updates the comments counter for a post.
- Added In Like the following method:
    A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class Comment < ApplicationRecord
+  belongs_to :user, class_name: 'User', foreign_key: 'AuthorId'
```
Comment (elisha2kyakpo1) [inline app/models/comment.rb]: - [ ] I like the association here between a user's table and comments table, but wait a bit, we are trying to create a custom table name attribute right? We want a user to be called the author (comment create). We can directly say that comment belongs to the author. 

Hint: `belongs_to :author, class_name: 'User'`

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  belongs_to :post, class_name: 'Post', foreign_key: 'PostId'
```
Comment (elisha2kyakpo1) [inline app/models/comment.rb]: - [ ] [OPTIONAL] The beauty of Rails is that it knows by default which foreign key names to give to the attributes being mapped to. For this line of code, when we say `belongs_to :post`, Rails will know that a comments table should have an attribute called `post_id`. We don't even need `class_name: 'Post', foreign_key: 'postId'`

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+  after_save :update_counter
+
+  def update_counter
+    post.CommentsCounter.nil? ? post.CommentsCounter = 1 : post.CommentsCounter += 1
+    post.update(CommentsCounter: post.CommentsCounter)
```
Comment (elisha2kyakpo1) [inline app/models/comment.rb]: - [ ] [OPTIONAL] I like the way you came up with these lines of code here, but have you thought about this Ruby helper method called `increment!`? Something like this: `author.increment!(:comments_counter)`

File: app/models/like.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class Like < ApplicationRecord
+  belongs_to :user, class_name: 'User', foreign_key: 'AuthorId'
+  belongs_to :post, class_name: 'Post', foreign_key: 'PostId'
+
+  after_save :update_counter
+
+  def update_counter
+    post.LikesCounter.nil? ? post.LikesCounter = 1 : post.LikesCounter += 1
+    post.update(LikesCounter: post.LikesCounter)
+  end
```
Comment (elisha2kyakpo1) [inline app/models/like.rb]: - [ ] Kindly use the same review for the comments model here as well.

File: app/models/post.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  belongs_to :user, class_name: 'User', foreign_key: 'AuthorId'
+  has_many :comments, class_name: 'Comment', foreign_key: 'PostId'
+  has_many :likes, class_name: 'Like', foreign_key: 'PostId'
+
+  after_save :update_counter
+
+  def update_counter
+    user.PostCounter.nil? ? user.PostCounter = 1 : user.PostCounter += 1
+    user.update(PostCounter: user.PostCounter)
+  end
```
Comment (elisha2kyakpo1) [inline app/models/post.rb]: - [ ] Kindly use the same review for the comments model here as well and remember that Ruby's naming convention for attribute names is camel_case.

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @bruk19, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- The PR title and description looks good and professional ✔️
- Descriptive commit messages ✔️
- No linter errors ✔️


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/comment.rb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'AuthorId'
```
Comment (danmainah) [inline app/models/comment.rb:2]: - [ ] [optional] You don't have to add a foreign key here as it's defined in the schema.

**Review by danmainah — APPROVED** [review]
Hi @bruk19 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights

✔️ No linter errors
✔️ Github flow was correctly used

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**